### PR TITLE
More Informative Scan: Threat List Page And Fix Dialogs

### PR DESCRIPTION
**Pull request by miguelxpn**
#### Changes proposed in this Pull Request

This updates the threat list page for the More Informative Scan project(pbuNQi-1IT-p2).

Design reference (p6TEKc-5tS-p2)

I'm not much of a front end developer, so I'm sure that I'm doing things wrong the scss side of things and there are some details I might have missed. Please pay extra attention to those items when reviewing 😅 

#### Testing instructions

* Set up a JP site with Scan and add a few threats (You can use the threat tester plugin to add a few innocuous threats for testing).
* Go to the scan section of both Calypso Blue and Calypso Green (To test Calypso Green you need to run yarn start start-jetpack-cloud and then go to http://jetpack.cloud.localhost:3000 on your browser) and check out the threat listing page.
* Verify that the layout matches the design specs linked above.
* I added the modifications behind a feature flag, as an additional test turn the flag off and verify that the design matches the current production design.
##### Before

Screen Shot 2021-11-22 at 23 14 38

##### After
Screen Shot 2021-11-22 at 23 25 29

### CODE REVIEW

File: client/components/jetpack/scan-threats/index.tsx [hunk 1]
```diff
@@ -51,6 +55,58 @@ const ScanError: React.FC< { site: Site } > = ( { site } ) => {
...
+				lowCount: numberFormat( countLowSeverity, 0 ),
+			},
+			comment: '$(lowCount)s is the number of low severity items found.',
+			count: countLowSeverity,
+		} );
+	}
+
+	let highThreatsSummary = '';
+	if ( countHighSeverity ) {
+		highThreatsSummary = translate( '%(threatCount)s threat', '%(threatCount)s threats', {
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( '%(threatCount)d threat found', '%(threatCount)d threats found' )` *ES Score: 7*

File: client/components/jetpack/scan-threats/index.tsx [hunk 2]
```diff
@@ -125,87 +181,124 @@ const ScanThreats = ( { error, site, threats }: Props ) => {
...
+	const countMapFixable = {
+		low: allFixableThreats.filter( ( threat ) => threat.severity < 3 ).length,
+		high: allFixableThreats.filter( ( threat ) => threat.severity >= 3 ).length,
+	};
+
+	const headerSummary = getThreatCountMessage(
+		countMap.high,
+		countMap.low,
+		null,
+		translate( 'found' )
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **39** times:
`translate( 'Found' )` *ES Score: 7*

File: client/components/jetpack/scan-threats/index.tsx [hunk 3]
```diff
@@ -125,87 +181,124 @@ const ScanThreats = ( { error, site, threats }: Props ) => {
...
+	);
+	const fixSummary = getThreatCountMessage(
+		countMapFixable.high,
+		countMapFixable.low,
+		translate( 'Jetpack can auto fix' ),
+		null
+	);
+
+	let headerMessage = translate( 'Your site is at risk' );
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Your site is at risk.' )` *ES Score: 10*

File: client/components/jetpack/scan-threats/index.tsx [hunk 4]
```diff
@@ -125,87 +181,124 @@ const ScanThreats = ( { error, site, threats }: Props ) => {
...
+					</FoldableCard>
+				</div>
 			</div>
 
 			{ ! error && (
 				<div className="scan-threats__rerun">
 					<p className="scan-threats__rerun-help">
 						{ translate(
-							'If you have manually fixed any of the threats above, you can {{button}}run a scan now{{/button}} or wait for Jetpack to scan your site later today.',
+							'If you have manually fixed any of the threats listed above, you can {{button}}run a manual scan now{{/button}} or wait for Jetpack to scan your site later today.',
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats/index.tsx]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

File: client/components/jetpack/threat-list-header/index.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import './style.scss';
+
+const ThreatListHeader = () => {
+	const translate = useTranslate();
+
+	return (
+		<CompactCard className="threat-list-header threat-list-header__wrapper">
+			<div className="threat-list-header__title title__severity">{ translate( 'Severity' ) }</div>
+			<div className="threat-list-header__title title__details">{ translate( 'Details' ) }</div>
+			<div className="threat-list-header__title title__autofix">{ translate( 'Auto Fix' ) }</div>
```
Comment (a8ci18n) [inline client/components/jetpack/threat-list-header/index.tsx:13]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Auto fix' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/more-informative-scan/threat-list)

**Comment by ederrengifo** [comment]
Awesome work! I have some feedback about small details to check and fix:

### (01) Be sure headers match the design
We made some adjustments in the header to add consistency across the different states. Right it seems different states are not sharing that new consistency. Check the [Figma file](https://www.figma.com/file/PB06PEvMkCtPaV4hFumLPQ/More-informative-Scan?node-id=1670%3A12996)

Screen Shot 2021-11-23 at 11 30 25

---

### (02) Dropdown arrows inside the low priority section don't match its status
When the low priority section is displayed, the arrows indicate that the low priority threats are already open when they are actually closed

Screen Shot 2021-11-23 at 11 23 06

---

### (03) Check the primary and secondary style of the buttons
Buttons should use primary and secondary styles of their own site (Calypso or Cloud). It seems right now we are overriding styles. Also, noticed how the green button (which in calypso should be as the other primary buttons) has a gray border.

Screen Shot 2021-11-23 at 11 22 37

---

### (04) Polish style for the code
The way we are displaying the code is a bit overwhelming. Maybe we could try reducing the font size?

Screen Shot 2021-11-23 at 11 23 31

---

### (05) If there is nothing to auto-fix, don't display the section
Right now it seems there is an empty area, it should be ok to hide the area and the line.

Screen Shot 2021-11-23 at 11 24 07

---

### (06) Check the font size used in sub-section titles
If you check the [design](https://www.figma.com/file/PB06PEvMkCtPaV4hFumLPQ/More-informative-Scan?node-id=1668%3A11997), these font sizes used for the titles should be the same

Screen Shot 2021-11-23 at 11 26 17

**Comment by miguelxpn** [comment]
> Awesome work! I have some feedback about small details to check and fix:
> 
> ### (01) Be sure headers match the design
> We made some adjustments in the header to add consistency across the different states. Right it seems different states are not sharing that new consistency. Check the [Figma file](https://www.figma.com/file/PB06PEvMkCtPaV4hFumLPQ/More-informative-Scan?node-id=1670%3A12996)
> 
> Screen Shot 2021-11-23 at 11 30 25
> 
> ### (02) Dropdown arrows inside the low priority section don't match its status
> When the low priority section is displayed, the arrows indicate that the low priority threats are already open when they are actually closed
> 
> Screen Shot 2021-11-23 at 11 23 06
> 
> ### (03) Check the primary and secondary style of the buttons
> Buttons should use primary and secondary styles of their own site (Calypso or Cloud). It seems right now we are overriding styles. Also, noticed how the green button (which in calypso should be as the other primary buttons) has a gray border.
> 
> Screen Shot 2021-11-23 at 11 22 37
> 
> ### (04) Polish style for the code
> The way we are displaying the code is a bit overwhelming. Maybe we could try reducing the font size?
> 
> Screen Shot 2021-11-23 at 11 23 31
> 
> ### (05) If there is nothing to auto-fix, don't display the section
> Right now it seems there is an empty area, it should be ok to hide the area and the line.
> 
> Screen Shot 2021-11-23 at 11 24 07
> 
> ### (06) Check the font size used in sub-section titles
> If you check the [design](https://www.figma.com/file/PB06PEvMkCtPaV4hFumLPQ/More-informative-Scan?node-id=1668%3A11997), these font sizes used for the titles should be the same
> 
> Screen Shot 2021-11-23 at 11 26 17

I fixed all the items! The scan header during the scanning process was modified in another PR, you can check it out [here](https://github.com/Automattic/wp-calypso/pull/57878).

File: client/components/jetpack/scan-threats-new/index.tsx [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+				lowCount: numberFormat( countLowSeverity, 0 ),
+			},
+			comment: '$(lowCount)s is the number of low severity items found.',
+			count: countLowSeverity,
+		} );
+	}
+
+	let highThreatsSummary = '';
+	if ( countHighSeverity ) {
+		highThreatsSummary = translate( '%(threatCount)s threat', '%(threatCount)s threats', {
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats-new/index.tsx:81]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( '%(threatCount)d threat found', '%(threatCount)d threats found' )` *ES Score: 7*

File: client/components/jetpack/scan-threats-new/index.tsx [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+	const countMapFixable = {
+		low: allFixableThreats.filter( ( threat ) => threat.severity < 3 ).length,
+		high: allFixableThreats.filter( ( threat ) => threat.severity >= 3 ).length,
+	};
+
+	const headerSummary = getThreatCountMessage(
+		countMap.high,
+		countMap.low,
+		null,
+		translate( 'found' )
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats-new/index.tsx:205]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **39** times:
`translate( 'Found' )` *ES Score: 7*

File: client/components/jetpack/scan-threats-new/index.tsx [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+	);
+	const fixSummary = getThreatCountMessage(
+		countMapFixable.high,
+		countMapFixable.low,
+		translate( 'Jetpack can auto fix' ),
+		null
+	);
+
+	let headerMessage = translate( 'Your site is at risk' );
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats-new/index.tsx:214]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Your site is at risk.' )` *ES Score: 10*

File: client/components/jetpack/scan-threats-new/index.tsx [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+							) ) }
+					</FoldableCard>
+				</div>
+			</div>
+
+			{ ! error && (
+				<div className="scan-threats-new__rerun">
+					<p className="scan-threats-new__rerun-help">
+						{ translate(
+							'If you have manually fixed any of the threats listed above, you can {{button}}run a manual scan now{{/button}} or wait for Jetpack to scan your site later today.',
```
Comment (a8ci18n) [inline client/components/jetpack/scan-threats-new/index.tsx:309]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'If you have manually fixed any of the threats above, you can {{button}}run a scan now{{/button}} or wait for Jetpack to scan your site later today.' )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/more-informative-scan/threat-list)

File: client/components/jetpack/threat-dialog-new/index.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+					? translate( 'Fix threat' )
+					: translate( 'Do you really want to ignore this threat?' ),
+			titleClassName: `threat-dialog-new__header--${ action }-threat`,
+		} ),
+		[ action ]
+	);
+
+	const getThreatSeverityText = ( threat ) => {
+		if ( threat.severity >= 5 ) {
+			return translate( 'critical' );
```
Comment (a8ci18n) [inline client/components/jetpack/threat-dialog-new/index.tsx:56]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **31** times:
`translate( 'Critical' )` *ES Score: 12*

**Review by bhoop — APPROVED** [review]
Works well for me!

One thing I noticed (that is far outside of the scope of this task) is that, because we don't associate threats found with the scans that found them, after all the threats are fixed the screen says "The last Jetpack scan ran XXX ago and everything looked great" even though the scan itself found threats. Something for us to think about.

**Review by bhoop — APPROVED** [review]
👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7039013

Thank you @miguelxpn for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7039013).

**Pull request merged**